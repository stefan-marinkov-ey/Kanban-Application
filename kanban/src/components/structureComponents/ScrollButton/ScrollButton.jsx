import React, { useState } from "react";
import { StyledScrollButton } from "./StyleScrollButton.style.jsx";

const ScrollButton = () => {
  const [visible, setVisible] = useState(false);

  const toggleVisible = () => {
    const scrolled = document.documentElement.scrollTop;
    if (scrolled > 300) {
      setVisible(true);
    } else if (scrolled <= 300) {
      setVisible(false);
    }
  };
  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  window.addEventListener("scroll", toggleVisible);

  return (
    <StyledScrollButton style={{ visibility: !visible && "hidden" }}>
      <span onClick={scrollToTop} style={{ visibility: !visible && "hidden" }}>
        ^
      </span>
    </StyledScrollButton>
  );
};

export default ScrollButton;
