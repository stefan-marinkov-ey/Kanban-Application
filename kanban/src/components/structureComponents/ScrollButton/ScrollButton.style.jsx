import styled from "styled-components";

export const Content = styled.div`
  overflow-y: scroll;
  height: 100%;
`;

export const StyledScrollButton = styled.div`
  position: fixed;
  width: 40px;
  right: 20px;
  bottom: 20px;
  height: 40px;
  border-radius: 50%;
  font-size: 3rem;
  z-index: 1;
  cursor: pointer;
`;
