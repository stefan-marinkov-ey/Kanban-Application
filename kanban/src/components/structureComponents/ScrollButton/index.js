import ScrollButton from "./ScrollButton";

export default ScrollButton;
export { ScrollButton };
