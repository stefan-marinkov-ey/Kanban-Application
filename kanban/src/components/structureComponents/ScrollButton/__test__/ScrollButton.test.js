import React from "react";
import { shallow } from "enzyme";
import ScrollButton from "../ScrollButton";

describe("ScrollButton", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ScrollButton />);
  });
  it("render component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
