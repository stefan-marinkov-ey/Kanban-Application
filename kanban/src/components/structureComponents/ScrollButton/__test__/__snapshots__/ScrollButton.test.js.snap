// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollButton render component 1`] = `ShallowWrapper {}`;
